components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/APIImportExportManager.java

```diff
@@ -59,19 +64,25 @@ public APIProvider getApiProvider() {
     /**
      * This method is used to export the given API as an archive (zip file).
      *
-     * @param apiToReturn       Requested API to export
-     * @param isStatusPreserved Is API status preserved or not
-     * @param exportFormat      Export file format of the API
-     * @return Archive file for the requested API
-     * @throws APIImportExportException If an error occurs while exporting the API and creating the archive
+     * @param apiTypeWrapper    Requested API or API Product to export
+     * @param isStatusPreserved Is API or API Product  status preserved or not
+     * @param exportFormat      Export file format of the API or the API Product
+     * @return Archive file for the requested API or API Product
+     * @throws APIImportExportException If an error occurs while exporting the API or the API Product and creating the archive
      */
-    public File exportAPIArchive(API apiToReturn, boolean isStatusPreserved, ExportFormat exportFormat)
+    public File exportAPIOrAPIProductArchive(ApiTypeWrapper apiTypeWrapper, boolean isStatusPreserved, ExportFormat exportFormat)
             throws APIImportExportException {
-
-        APIIdentifier apiIdentifier = apiToReturn.getId();
-        String archiveBasePath = exportAPIArtifacts(apiToReturn, isStatusPreserved, exportFormat);
+        String archiveBasePath;
+        if (!apiTypeWrapper.isAPIProduct()) {
+            APIIdentifier apiIdentifier = apiTypeWrapper.getApi().getId();
+            archiveBasePath = exportAPIArtifacts(apiTypeWrapper.getApi(), isStatusPreserved, exportFormat);
+            log.info("API" + apiIdentifier.getApiName() + "-" + apiIdentifier.getVersion() + " exported successfully");
```

Comment:
Do we need this as an info log? Shall we make this "debug"?

Reply:
Sure will change it here and in line 83 as well.